addons/product/models/product_packaging.py

```diff
@@ -58,7 +58,9 @@ def _find_suitable_product_packaging(self, product_qty, uom_id):
         """ try find in `self` if a packaging's qty in given uom is a divisor of
         the given product_qty. If so, return the one with greatest divisor.
         """
-        packagings = self.sorted(lambda p: p.qty, reverse=True)
+        packagings = self\
+                .filtered(lambda p: p.product_id.company_id <= p.company_id <= p.env.companies)\
```

Comment:
`_find_suitable_product_packaging` should receive the company to filter on by the object calling it. If you have access to company1 and company2. Both of them contains a different packaging. If the method is called from a stock.move in company2, we should not return the company1's packaging 

Also `p.product_id.company_id <= p.company_id` is already satisfied by the `check_company` arg in the `product_id` field definition

Reply:
> `_find_suitable_product_packaging` should receive the company to filter on by the object calling it. If you have access to company1 and company2. Both of them contains a different packaging. If the method is called from a stock.move in company2, we should not return the company1's packaging

Do you mean access as in administrative access, or a multicompany environment where you have several companies selected in the dropdown menu? If it's the former, then yes, that is the issue this fixes. In case of the latter, I think it's logical that if you're working with multiple companies simultaneously in one environment, it can select packaging from any of those companies, no?

> Also p.product_id.company_id <= p.company_id is already satisfied by the check_company arg in the product_id field definition

At the moment of creation, yes, but there's nothing stopping you from invalidating that afterwards by modifying the product's `company_id`.